Feature/timechanger

williamwinkler opened the pull request:
Added TimeChanger feature

williamwinkler committed:
Feature added: Extend cooking time
This feature adds 30 seconds to the timer, if the microwave is cooking

williamwinkler committed:
Feature added: Extend cooking time
This feature adds 30 seconds to the timer, if the microwave is cooking

williamwinkler committed:
Added tests of TimeChanger

williamwinkler committed:
Added tests of TimeChanger

JacobBrejnbjerg commented on Microwave.Test.Unit/TimerTest.cs:
```diff
@@ -147,5 +147,29 @@ public void Tick_Started_TimeRemainingCorrect(int ticks)
             Assert.That(uut.TimeRemaining, Is.EqualTo(5-ticks*1));
         }
+
+        [TestCase(1, 0)]
```
Det vil også være en god idé at teste med negative værdier

williamwinkler replied:
Ny test med negative værdier er tilføjet, AddTime kaster en exception på negative værdier

JacobBrejnbjerg commented on Microwave.Test.Unit/UserInterfaceTest.cs:
```diff
@@ -41,6 +41,24 @@ public void Setup()
+            startCancelButton.Pressed += Raise.EventWith(this, EventArgs.Empty);
+
+            // Add 30 seconds to timer by pressing the timer button during cooking
+            timeButton.Pressed += Raise.EventWith(this, EventArgs.Empty);
```
Lav det til et for-loop. Så kan man verificere, at det fungere korrekt hvis man klikker mere end 1 gang

williamwinkler replied:
Forloop tilføjet, så det er nu verficeret, at knappen kan trykkes flere gange og cooker bliver kaldt det antal gange, som er blevet trykket.
Hvis knappen ikke trykkes, så sker der heller intet -> testet og OK.

Commit:
Added additional test to verify that CookController reads correct MaxPower correctly from the PowerTube

williamwinkler committed:
Reworked AddTime

now throws an error when seconds is negative

williamwinkler committed:
Reworked AddTime

now throws an error when seconds is negative

williamwinkler committed:
TimeChanger: Added more tests
AddTime kan ikke være negativ, og der testes på om knappen fungerer ved flere klik også

williamwinkler committed:
TimeChanger: Added more tests
AddTime kan ikke være negativ, og der testes på om knappen fungerer ved flere klik også

williamwinkler committed:
No press on timeButton results in no time added during cooking -> testet!

williamwinkler committed:
No press on timeButton results in no time added during cooking -> testet!

williamwinkler committed:
Merge branch 'feature/timechanger' of https://github.com/JacobBrejnbjerg/SWT_Microwave_H3 into feature/timechanger

JacobBrejnbjerg reviewed (approved):
Godt med de nye tilføjelser til tests samt at AddTime i CookControlleren nu kun tager sekunder og ikke minutter som parameter.
